contracts/extensions/bridge-operator-governance/BridgeManager.sol

```diff
@@ -0,0 +1,503 @@
+  }
+
+  /**
+   * @inheritdoc IHasContracts
+   */
+  function setContract(ContractType contractType, address addr) external override onlySelfCall {
+    _requireHasCode(addr);
+    _setContract(contractType, addr);
+  }
+
+  /**
+   * @inheritdoc IQuorum
+   */
+  function setThreshold(
+    uint256 numerator,
+    uint256 denominator
+  ) external override onlySelfCall returns (uint256, uint256) {
+    return _setThreshold(numerator, denominator);
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function getTotalWeights() public view returns (uint256) {
+    return TOTAL_WEIGHTS_SLOT.load();
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function getGovernorWeights(address[] calldata governors) external view returns (uint256[] memory weights) {
+    weights = _getGovernorWeights(governors);
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function sumGovernorsWeight(address[] memory governors) public view nonDuplicate(governors) returns (uint256 sum) {
+    uint256 length = _getBridgeOperatorSet().length();
+    mapping(address => BridgeOperatorInfo) storage _governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+    for (uint256 i; i < length; ) {
+      sum += _governorToBridgeOperatorInfo[governors[i]].voteWeight;
+
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function totalBridgeOperators() external view returns (uint256) {
+    return _getBridgeOperatorSet().length();
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function isBridgeOperator(address addr) external view returns (bool) {
+    return _getBridgeOperatorSet().contains(addr);
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function getBridgeOperators() external view returns (address[] memory) {
+    return _getBridgeOperators();
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function getGovernors() external view returns (address[] memory) {
+    return _getGovernors();
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function getBridgeOperatorOf(address[] calldata governors) external view returns (address[] memory bridgeOperators_) {
+    uint256 length = governors.length;
+    bridgeOperators_ = new address[](length);
+
+    mapping(address => BridgeOperatorInfo) storage _gorvernorToBridgeOperator = _getGovernorToBridgeOperatorInfo();
+    for (uint256 i; i < length; ) {
+      bridgeOperators_[i] = _gorvernorToBridgeOperator[governors[i]].addr;
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  function getGovernorsOf(address[] calldata bridgeOperators) external view returns (address[] memory governors) {
+    uint256 length = bridgeOperators.length;
+    governors = new address[](length);
+    mapping(address => address) storage _governorOf = _getGovernorOf();
+
+    for (uint256 i; i < length; ) {
+      governors[i] = _governorOf[bridgeOperators[i]];
+      unchecked {
+        ++i;
+      }
+    }
+  }
```

Comment:
I cannot see any loc that uses these methods

Reply:
I think it maybe useful for future usage.